ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -2089,6 +2089,25 @@ IHqlExpression * ThorHqlTransformer::createTransformed(IHqlExpression * expr)
     case no_debug_option_value:
         //pick best engine etc. definitely done by now, so substitute any options that haven't been processed already
         return getDebugValueExpr(translator.wu(), expr);
+    case no_embedbody:
+        {
+            //Convert all definitions of non functional embeds to implicit functions
+            StringBuffer funcname;
+            funcname.append("userx").append(++implicitFunctionId);
+            HqlExprArray args;
+            args.append(*LINK(expr));
+            if (expr->hasAttribute(languageAtom))
+            {
+                args.append(*createAttribute(contextAtom));
```

Comment:
yes.  This code was adapted from code in hqlgram2.cpp.  The contextAtom ensure that the ICodeContext is passed to the generated function.  The generated code for non-c++ functions uses the code context in the code that is generated.